src/builtin.js

```diff
@@ -908,70 +566,73 @@ Sk.builtin.hash = function hash (value) {
+    } else if (typeof value === "number" || value === null || value === true || value === false) {
         throw new Sk.builtin.TypeError("unsupported Javascript type");
     }
 
-    return new Sk.builtin.str((typeof value) + " " + String(value));
+    return new Sk.builtin.str(typeof value + " " + String(value));
     // todo; throw properly for unhashable types
 };
 
-Sk.builtin.getattr = function getattr (obj, pyName, default_) {
-    var ret, mangledName, jsName;
-    Sk.builtin.pyCheckArgsLen("getattr", arguments.length, 2, 3);
+Sk.builtin.getattr = function getattr(obj, pyName, default_) {
     if (!Sk.builtin.checkString(pyName)) {
         throw new Sk.builtin.TypeError("attribute name must be string");
     }
-
-    jsName = pyName.$jsstr();
-    mangledName = new Sk.builtin.str(Sk.fixReservedWords(jsName));
-    ret = obj.tp$getattr(mangledName);
-    if (ret === undefined) {
-        if (default_ !== undefined) {
-            return default_;
-        } else {
-            throw new Sk.builtin.AttributeError("'" + Sk.abstr.typeName(obj) + "' object has no attribute '" + jsName + "'");
+    const res = Sk.misceval.tryCatch(
+        () => obj.tp$getattr(pyName, true),
+        (e) => {
+            if (e instanceof Sk.builtin.AttributeError) {
+                return undefined;
+            } else {
+                throw e;
+            }
         }
-    }
-    return ret;
+    );
+    return Sk.misceval.chain(res, (r) => {
+        if (r === undefined) {
+            if (default_ !== undefined) {
+                return default_;
+            }
+            throw new Sk.builtin.AttributeError("'" + Sk.abstr.typeName(obj) + "' object has no attribute " + pyName.$jsstr());
+        }
+        return r;
+    });
 };
 
-Sk.builtin.setattr = function setattr (obj, pyName, value) {
-    var jsName;
-    Sk.builtin.pyCheckArgsLen("setattr", arguments.length, 3, 3);
+Sk.builtin.setattr = function setattr(obj, pyName, value) {
     // cannot set or del attr from builtin type
     if (!Sk.builtin.checkString(pyName)) {
         throw new Sk.builtin.TypeError("attribute name must be string");
     }
-    jsName = pyName.$jsstr();
-    if (obj.tp$setattr) {
-        obj.tp$setattr(new Sk.builtin.str(Sk.fixReservedWords(jsName)), value);
-    } else {
-        throw new Sk.builtin.AttributeError("object has no attribute " + jsName);
-    }
-    return Sk.builtin.none.none$;
+    const res = obj.tp$setattr(pyName, value, true);
+    return Sk.misceval.chain(res, () => {
+        return Sk.builtin.none.none$;
+    });
 };
 
 Sk.builtin.raw_input = function (prompt) {
     var lprompt = prompt ? prompt : "";
 
     return Sk.misceval.chain(Sk.importModule("sys", false, true), function (sys) {
         if (Sk.inputfunTakesPrompt) {
-            return Sk.misceval.callsimOrSuspendArray(Sk.builtin.file.$readline, [sys["$d"]["stdin"], null, lprompt]);
+            return Sk.builtin.file.$readline(sys["$d"]["stdin"], null, lprompt);
```

Comment:
Chain handles the suspension already. Here we directly call the function since we can. $readline is a js function and so no reason to use callsim here.